src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6393,8 +6399,53 @@ void QueryAnalyzer::resolveTableFunction(QueryTreeNodePtr & table_function_node,
     QueryExpressionsAliasVisitor & expressions_visitor,
     bool nested_table_function)
 {
+
+    String database_name = scope.context->getCurrentDatabase();
+    String table_name;
+
+    if (table_function_node->getOriginalAST() && table_function_node->getOriginalAST()->as<ASTFunction>())
```

Comment:
`getOriginalAST` is used 5 times in this scope, let's save its value into a local variable and use it? or better save the value of `table_function_node->getOriginalAST()->as<ASTFunction>()` as it is used 4 times

Reply:
addressed it